### DISCUSSION TITLE
2022-12-08のミーティングアジェンダ

### DISCUSSION THREAD

**Discussion by azu** [asciidwango/js-primer#1542] [Meeting]
2022年12月8日のミーティング用

**Comment by azu** [comment]
- https://github.com/asciidwango/js-primer/issues/1528

ウェブ版はただのリンク https://jsprimer.net/intro/#changelog
書籍版は次のような感じ

----

初版からの変更点をまとめると次のようになります。

- ECMAScriptの新しいバージョンであるES2020、ES2021、ES2022に対応した
- 新しいECMAScriptの機能によって、使う必要がなくなった機能は非推奨へと変更した
- 文字では想像しにくいビット演算、非同期処理などに図を追加した
- PromiseとAsync Functionを非同期の処理の中心として書き直した
- 一方で、エラーファーストコールバックは非同期処理としてはメインではなくなった
- `Array#includes`という表記は、Private Classs Fields(`#field`)と記号が被るため廃止した
- Node.jsでもECMAScript Modulesを使うようになり、CommonJSはメインではなくなった
- Node.jsが12から18までアップデートし、npmは6から8までアップデート、各種ライブラリも最新にアップデート
- 読者からのフィードバックを受けて、全体をより分かりやすく読みやすくなるように書き直した

**Comment by azu** [comment]
[refactor(use-case/ajaxapp): サンプルコードのディレクトリをexampleに移動 by lacolaco · Pull Request #1538 · asciidwango/js-primer](https://github.com/asciidwango/js-primer/pull/1538)

**Reply by lacolaco** [reply]
今作業中　src消して `example` の下にHTMLとJSを置いた状態にする

**Reply by lacolaco** [reply]
作業done mergeすれば終わり

**Comment by azu** [comment]
https://github.com/nodejs/Release/issues/778#issuecomment-1265458751
     
```
Wednesday Jan. 18th (~6 weeks after v19 backport)
    PR is open to backport npm@9.x in node@18
```

これ間に合うかどうか
2月前ぐらいにぎりぎりLTSにnpm9が入るイメージ
